pkg/cri/server/service.go

```diff
@@ -165,8 +167,15 @@ func NewCRIService(config criconfig.Config, options *CRIServiceOptions) (CRIServ
 		sandboxService:     newCriSandboxService(&config, options.Client),
 	}
 
-	// TODO: figure out a proper channel size.
-	c.containerEventsChan = make(chan runtime.ContainerEventResponse, 1000)
+	// TODO: Make discard time configurable
+	c.containerEventsQ = eventq.New[runtime.ContainerEventResponse](5*time.Minute, func(event runtime.ContainerEventResponse) {
```

Comment:
Curious; with the work being done on OTEL, and with context-loggers being used; would it make sense to have these signatures accept a context? (code below could use the context-logger, and perhaps the context would be useful to pass around?

Reply:
Add context to the `New` interface or callback or both? There isn't much inside the event queue itself that would would care about context or be able to manage it. I ended up adding a `Shutdown` to allows tests to synchronization, I don't think context cancellation is better for that. It seems to me though that context should either be embedded in the event types or part of the callback.